app/src/main/java/by/paranoidandroid/cityweather/view/fragment/CitiesFragment.kt

```diff
@@ -4,72 +4,85 @@ import android.arch.lifecycle.Observer
 import by.paranoidandroid.cityweather.R
-import by.paranoidandroid.cityweather.Utils.TAG
-import by.paranoidandroid.cityweather.domain.entity.Coord
-import by.paranoidandroid.cityweather.domain.entity.Forecast
-import by.paranoidandroid.cityweather.domain.entity.Main
+import by.paranoidandroid.cityweather.Utils.LOG_TAG
 import by.paranoidandroid.cityweather.view.LoadingView
 import by.paranoidandroid.cityweather.view.adapter.CityAdapter
 import by.paranoidandroid.cityweather.viewmodel.CitiesViewModel
 import by.paranoidandroid.cityweather.viewmodel.CitiesViewModelFactory
-import kotlinx.android.synthetic.main.fragment_cities.*
-import kotlinx.android.synthetic.main.fragment_cities.view.*
 import javax.inject.Inject
 
 class CitiesFragment: Fragment(), LoadingView {
     @Inject
     lateinit var viewModelFactory: CitiesViewModelFactory
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        val view = inflater.inflate(R.layout.fragment_cities, container, false)
-
-        val cityAdapter = CityAdapter(context)
-        view.recyclerViewCities.adapter = cityAdapter
-        view.recyclerViewCities.layoutManager = LinearLayoutManager(context)
+    lateinit var viewModel: CitiesViewModel
+    private var cityAdapter: CityAdapter? = null
+    private var recyclerViewCities: RecyclerView? = null
+    private var progressBar: ContentLoadingProgressBar? = null
+    private var tvError: TextView? = null
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
         AndroidApplication.injector.inject(this)
-        val viewModel = ViewModelProviders.of(this, viewModelFactory).get(CitiesViewModel::class.java)
-        viewModel.getForecasts()?.observe(this, object : Observer<List<Forecast<Main, Coord>>> {
-            override fun onChanged(cityList: List<Forecast<Main, Coord>>?) {
-                Log.d(TAG, "onChanged in CitiesFragment")
-                if (cityList != null) {
-                    Log.d(TAG, "onChanged and cityList != null")
-                    cityAdapter.updateItems(cityList)
-                }
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+                                      .get(CitiesViewModel::class.java)
+        cityAdapter = CityAdapter(context)
+
+        viewModel.getForecasts().observe(this, Observer<ForecastList?> { cityList ->
+            if (cityList != null) {
+                cityAdapter?.updateItems(cityList)
             }
         })
+    }
+
+    override fun onCreateView(inflater: LayoutInflater,
+                              container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_cities, container, false)
+        recyclerViewCities = view.findViewById(R.id.recycler_view_cities)
+        recyclerViewCities?.layoutManager = LinearLayoutManager(context)
```

Comment:
move to xml

Reply:
✅ Fixed in commits 7ef84fc248a708705edd94039029d274d7f70c50 (_"Fix of previous commit (RetrofitModule was renamed to NetworkModule) and refactoring"_)
`LayoutManager ` already was defined in xml:

```
<android.support.v7.widget.RecyclerView
        android:id="@+id/recycler_view_cities"
       ...
        app:layoutManager="android.support.v7.widget.LinearLayoutManager"
        ... />
```
so I've just removed this unnecessary line in code:

`recyclerViewCities?.layoutManager = LinearLayoutManager(context)`